packages/solid-query/src/QueryClientProvider.tsx

```diff
@@ -55,56 +25,26 @@ export const useQueryClient = ({ context }: ContextOptions = {}) => {
   return queryClient
 }
 
-type QueryClientProviderPropsBase = {
+export type QueryClientProviderProps = {
   client: QueryClient
   children?: JSX.Element
-}
-type QueryClientProviderPropsWithContext = ContextOptions & {
-  contextSharing?: never
-} & QueryClientProviderPropsBase
-type QueryClientProviderPropsWithContextSharing = {
-  context?: never
-  contextSharing?: boolean
-} & QueryClientProviderPropsBase
-
-export type QueryClientProviderProps =
-  | QueryClientProviderPropsWithContext
-  | QueryClientProviderPropsWithContextSharing
+} & ContextOptions
 
-export const QueryClientProvider = (
-  props: QueryClientProviderProps,
-): JSX.Element => {
-  const mergedProps = mergeProps(
-    {
-      contextSharing: false,
-    },
-    props,
-  )
+export const QueryClientProvider = ({
+  client,
```

Comment:
I think that `destructuring` props in solid will break the reactivity.
@ardeora

Reply:
You might be right
https://www.solidjs.com/tutorial/props_split

Reply:
Oh, thanks for pointing this out. I've never used solid, I'll put it back to how it was :)